v2/api/compute/v1api20201201/structure.txt

```diff
@@ -1866,6 +1866,62 @@ github.com/Azure/azure-service-operator/v2/api/compute/v1api20201201
 │   │   └── Tier: *string
 │   ├── Tags: map[string]string
 │   └── Zones: string[]
+├── VirtualMachineScaleSetsExtension: Resource
+│   ├── Owner: VirtualMachineScaleSet
+│   ├── Spec: Object (11 properties)
+│   │   ├── AutoUpgradeMinorVersion: *bool
+│   │   ├── AzureName: string
+│   │   ├── EnableAutomaticUpgrade: *bool
+│   │   ├── ForceUpdateTag: *string
+│   │   ├── Owner: *genruntime.KnownResourceReference
+│   │   ├── ProtectedSettings: map[string]v1.JSON
```

Comment:
https://github.com/Azure/azure-quickstart-templates/blob/master/quickstarts/microsoft.compute/vmss-trustedlaunch-windows/azuredeploy.json#L400

Yes, it's nested.

Reply:
So I think we have a few options here:
1. We can force-transform this into a `map[string]string` and then use the new `SecretMatReference` approach I just merged. CAPZ actually treats/documents this as a `map[string]string` as you can see [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/c1e35e3be57c930e53f5d7c0736f7264e1713332/docs/book/src/topics/custom-vm-extensions.md?plain=1#L26). This isn't _great_ because it's not the full representation the API allows but it is a subset and we could probably expand it in a nonbreaking way in the future (by supporting arbitrary JSON in the secret and rendering it into JSON in the request. My assumption is that this is what the `ProtectedSettingsFromKeyVault` which was added in the newer APIs is already doing w/ KV secrets.
2. We can just strip this field out entirely for now and file an item to bring it back.

Whatever we do, we _probably_ should take a breaking change to do it for the Extension embedded in VM/VMSS too?

Reply:
I think 2 makes sense for now, I'll open a new issue and we can discuss on the approach we wanna take for this.